skyportal/handlers/api/observingrun.py

```diff
@@ -0,0 +1,226 @@
+
+        DBSession().add(run)
+        DBSession().commit()
+
+        return self.success(data={"id": run.id})
+
+    @auth_or_token
+    def get(self, run_id=None):
+        """
+        ---
+        single:
+          description: Retrieve an observing run
+          parameters:
+            - in: path
+              name: run_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleObservingRunGetWithAssignments
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all observing runs
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfObservingRunGets
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+        if run_id is not None:
+            run = DBSession().query(ObservingRun).options(
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.obj)
+                .joinedload(Obj.thumbnails)
+                .joinedload(Thumbnail.photometry),
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.requester),
+                joinedload(ObservingRun.assignments)
+                .joinedload(ClassicalAssignment.obj)
+                .joinedload(Obj.comments),
+            ).filter(ObservingRun.id == run_id).first()
+
+            if run is None:
+                return self.error(f"Could not load observing run {run_id}",
+                                  data={"run_id": run_id})
+            # order the assignments by ra
+            run.assignments = sorted(
+                run.assignments, key=lambda a: a.obj.ra
+            )
+
+            # filter out the assignments of objects that are not visible to
+            # the user
+            run.assignments = list(filter(
+                lambda a: a.obj.is_owned_by(self.current_user),
+                run.assignments
+            ))
+
+            for assignment in run.assignments:
+                assignment.obj.comments = sorted(assignment.obj.comments,
+                                                 key=lambda c: c.modified,
+                                                 reverse=True)
+
+            data = ObservingRunGetWithAssignments.dump(run)
+            data['assignments'] = [a.to_dict() for a in data['assignments']]
+
+            # calculate when the targets rise and set
+            for d, a in zip(data['assignments'], run.assignments):
+                d['rise_time_utc'] = a.rise_time.isot
+                d['set_time_utc'] = a.set_time.isot
+
+            return self.success(data=data)
+        runs = ObservingRun.query.all()
```

Comment:
I think I may have asked this before, but can anyone view all observing runs?

Reply:
Yes

Reply:
Telescope schedules are public